autogluon/utils/tabular/ml/models/xgboost/callbacks.py

```diff
@@ -0,0 +1,167 @@
+
+        state['maximize_score'] = maximize_score
+        state['best_iteration'] = 0
+        if maximize_score:
+            state['best_score'] = float('-inf')
+        else:
+            state['best_score'] = float('inf')
+        msg = '[%d]\t%s' % (
+            env.iteration,
+            '\t'.join([_fmt_metric(x) for x in env.evaluation_result_list]))
+        state['best_msg'] = msg
+
+        if bst is not None:
+            if bst.attr('best_score') is not None:
+                state['best_score'] = float(bst.attr('best_score'))
+                state['best_iteration'] = int(bst.attr('best_iteration'))
+                state['best_msg'] = bst.attr('best_msg')
+            else:
+                bst.set_attr(best_iteration=str(state['best_iteration']))
+                bst.set_attr(best_score=str(state['best_score']))
+        else:
+            assert env.cvfolds is not None
+
+    def callback(env):
+        """internal function"""
+        if not state:
+            init(env)
+        score = env.evaluation_result_list[-1][1]
+        best_score = state['best_score']
+        best_iteration = state['best_iteration']
+        maximize_score = state['maximize_score']
+        
+        if (maximize_score and score > best_score) or \
+                (not maximize_score and score < best_score):
+            msg = '[%d]\t%s' % (
+                env.iteration,
+                '\t'.join([_fmt_metric(x) for x in env.evaluation_result_list]))
+            state['best_msg'] = msg
+            state['best_score'] = score
+            state['best_iteration'] = env.iteration
+            # save the property to attributes, so they will occur in checkpoint.
+            if env.model is not None:
+                env.model.set_attr(best_score=str(state['best_score']),
+                                   best_iteration=str(state['best_iteration']),
+                                   best_msg=state['best_msg'])
+        elif env.iteration - best_iteration >= stopping_rounds:
+            best_msg = state['best_msg']
+            if verbose and env.rank == 0:
+                msg = "Stopping. Best iteration:\n{}\n\n"
+                rabit.tracker_print(msg.format(best_msg))
+            raise EarlyStopException(best_iteration)
+
+        if env.iteration % 10 == 0:
+            available = psutil.virtual_memory().available
+            cur_rss = mem_status.memory_info().rss
+            # rabit.tracker_print(f"{state}\n")
+            # rabit.tracker_print(f"CUR_RSS: {cur_rss}\n")
+            if cur_rss < state['init_mem_rss']:
+                state['init_mem_rss'] = cur_rss
+            estimated_model_size_mb = (cur_rss - state['init_mem_rss']) >> 20
+            available_mb = available >> 20
+
+            model_size_memory_ratio = estimated_model_size_mb / available_mb
+            if verbose and (model_size_memory_ratio > 0.25):
+                rabit.tracker_print(f'Available Memory: {available_mb} MB\n')
```

Comment:
Could you explain why we are using `rabit.tracker_print` instead of `logger.log`? I'm unfamiliar with `rabit`.

Reply:
`rabit` tracker is a default logging of xgboost library. I will change that with `logger.log`. Sorry for confusing.